airflow/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -617,6 +617,13 @@ def on_kill(self):
             self._submit_sp.kill()
 
             if self._yarn_application_id:
+                if self._keytab is not None and self._principal is not None:
+                    kinit_cmd = "kinit -kt {} {}" \
+                        .format(self._keytab, self._principal).split()
+                    kinit = subprocess.Popen(kinit_cmd,
+                                             stdout=subprocess.PIPE,
+                                             stderr=subprocess.PIPE)
```

Comment:
I think this should be

```suggestion
                    from airflow.security.kerberos import renew_from_kt
                    renew_from_kt(self._principal, self._keytab)
```

Reply:
(But I'm not familiar with Kerberos, so I could be wrong)

Reply:
cool, that would renew the kerberos token and store it at `/tmp/airflow_krb5_ccache`(configured in `airflow.cfg`)  but yarn expects it at `/tmp/krb5cc_<uid>` the default token location , It would work if the env variable `KRB5CCNAME` is set to `/tmp/<airflow.cfg.kerberos.ccache >` in the subprocess used for  yarn kill. will update it

Reply:
@ashb  - tried using the `renew_from_kt` and it seems to do the following

`kinit -r 3600m -k -t my.keytab -c /tmp/airflow_krb5_ccache principal`

and then does a workaround to renew the ticket

`kinit -c /tmp/airflow_krb5_ccache -R` 

the latter seems to fail  for non-renewable tickets and does a `sys.exit`, any suggestions on how this can be handled? For a yarn kill, the renewal work around won't be needed

Reply:
This change looks good, but can you add a comment to the nrew_from_kt call in SparkOperator to say why we aren't treating the errors as fatal.

Reply:
done, can you have another look please